module/mempool/queue/entityRequests.go

```diff
@@ -0,0 +1,103 @@
+package queue
```

Comment:
This is a lot of largely duplicated code and tests (~300 lines), to add a new `HeroCache` for the `EntityRequest` type. We would need to duplicate and maintain this code each time we use `HeroCache` as the queue backend for a different type. To me, this is not worth the trade-off here. 

Why don't we default to using `FifoQueue`, which supports generic entities and is used extensively for this role elsewhere, instead? Alternatively, we could consider modify `HeroCache` to support generic entities.

Reply:
Generalized `HeroStore` and `HeroQueue` to support `engine.Message` and `flow.Entity`, respectively in https://github.com/onflow/flow-go/pull/3404/commits/73825e4dc2a8997ba93c5ac4876c4454084b6e8a, and replaced the `ChunkDataPackRequestStore` and `EntityRequestStore` with `HeroStore`.